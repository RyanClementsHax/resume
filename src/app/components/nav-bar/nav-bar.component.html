<nav class="nav-bar">
  <a class="nav-button">Ryan Clements</a>
  <button
    type="button"
    aria-label="Toggle menu"
    class="nav-toggle nav-button"
    [ngClass]="expanded ? 'expanded' : ''"
    (click)="toggleMenu()">
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 30 30"
      width="30"
      height="30"
      focusable="false">
      <title>Menu</title>
      <path
        stroke="currentColor"
        stroke-width="2"
        stroke-linecap="round"
        stroke-miterlimit="10"
        d="M4 7h22M4 15h22M4 23h22">
      </path>
    </svg>
  </button>
  <ul class="nav-links" [ngClass]="expanded ? 'expanded' : ''">
    <li class="nav-button nav-link">Item 1</li>
    <li class="nav-button nav-link">Item 2</li>
    <li class="nav-button nav-link">Item 3</li>
  </ul>
</nav>