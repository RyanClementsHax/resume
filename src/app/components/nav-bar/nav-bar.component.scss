@import "../../../variables.scss";

$button-border-right: 5px;

:host {
    width: 100%;
    height: $nav-bar-height;
    background: $color-background-main;

    .nav-bar {
        display: flex;
        flex-direction: row;
        justify-content: flex-start;
        border-bottom: $light-border;
    
        .small-nav-container {
            display: flex;
        }
    
        .nav-links {
            margin-top: 0;
            margin-left: auto;
            margin-bottom: 0;
            padding: 0;
        }
    
        .link-container {
            cursor: pointer;
            display: inline-block;
            background: $color-background-main;
        }
    
        .nav-button {
            display: block;
            font-weight: 700;
            text-transform: uppercase;
            letter-spacing: .2em;
            padding: $text-padding;
            -webkit-appearance: none;

            &:visited {
                color: inherit;
            }
        }
        
        .nav-button:hover {
            color: $color-primary;
            background-color: $color-primary-faded;
            border-right: $button-border-right solid $color-accent;
            padding-right: calc(#{$text-padding} - #{$button-border-right});
        }
    
        .nav-toggle {
            cursor: pointer;
            display: none;
            -webkit-user-select: none;
            -moz-user-select: none;
            -ms-user-select: none;
            user-select: none;
            margin-left: auto;
            border: 0;
            padding: 0 $text-padding;
            background: $color-background-main;
    
            svg {
                width: 2rem;
                height: 2rem;
            }
        }
    
        .nav-toggle.expanded {
            background: $color-primary-faded;
        }
    }
    
    @media screen and (max-width: 45em) {
        .nav-bar {
            flex-direction: column;
    
            .nav-links {
                display: flex;
                flex-direction: column;
                width: 100%;
                overflow: hidden;
                transition: max-height 250ms;
    
                .link-container {
                    border-top: $light-border;
        
                    .nav-link {
                        width: 100%;
                    }
                }
            }
        
            .nav-toggle {
                display: block;
            }
        }
    }
}
