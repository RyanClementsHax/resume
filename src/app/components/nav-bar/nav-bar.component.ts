import { Component, AfterViewInit, ViewChild, ElementRef } from '@angular/core';
import { Router, NavigationStart, Event } from '@angular/router';

import { filter } from 'rxjs/operators';

@Component({
  selector: 'nav-bar',
  templateUrl: './nav-bar.component.html',
  styleUrls: ['./nav-bar.component.scss']
})
export class NavBarComponent implements AfterViewInit {
  @ViewChild('links', { static: false }) links: ElementRef;

  expanded = false;
  linksMaxHeight: number;

  constructor(private router: Router) {
    this.router.events.pipe(
        filter((event: Event) => event instanceof NavigationStart)
      )
      .subscribe(() => {
        this.expanded = false;
      });
  }

  ngAfterViewInit() {
    const linkHeight = this.links.nativeElement.childNodes[0].offsetHeight;
    const linkCount = this.links.nativeElement.childNodes.length;
    this.linksMaxHeight = linkCount * linkHeight;
  }

  toggleMenu() {
    this.expanded = !this.expanded;
  }
}
