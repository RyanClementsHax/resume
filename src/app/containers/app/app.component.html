<nav-bar></nav-bar>
<router-outlet></router-outlet>
<footer>
    <div class="copyRight">
        &copy; Ryan Clements
    </div>
</footer>
