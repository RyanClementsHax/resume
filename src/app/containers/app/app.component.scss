@import "../../../variables.scss";

:host {
    width: 100%;
    position: relative;
    padding-top: $nav-bar-height;

    nav-bar {
        position: absolute;
        top: 0;
    }

    footer {
        height: 50px;
        border-top: $light-border;
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: left;

        .copyRight {
            padding-left: 1rem;
        }
    }
}