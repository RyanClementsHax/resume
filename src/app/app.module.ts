import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';

import { AppComponent } from './containers/app/app.component';
import { NavBarComponent } from './components/nav-bar/nav-bar.component';

@NgModule({
  declarations: [
    AppComponent,
    NavBarComponent
  ],
  imports: [
    BrowserModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
