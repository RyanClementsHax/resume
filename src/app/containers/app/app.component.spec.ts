import { TestBed, async } from '@angular/core/testing';
import { AppComponent } from './app.component';
import { NavBarComponent } from '../../components/nav-bar/nav-bar.component';
describe('AppComponent', () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        AppComponent,
        NavBarComponent
      ],
    }).compileComponents();
  }));

  it('should create the app', async(() => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.debugElement.componentInstance;
    expect(app).toBeTruthy();
  }));
});
