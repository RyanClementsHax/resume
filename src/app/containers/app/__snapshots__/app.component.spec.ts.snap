// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match snapshot 1`] = `AppComponent {}`;
